urllib3/util/retry.py

```diff
@@ -330,7 +338,7 @@ def increment(self, method=None, url=None, response=None, error=None,
 
         elif error and self._is_read_error(error):
             # Read retry?
-            if read is False:
+            if read is False or (method is not None and not self._is_method_retryable(method)):
```

Comment:
Ah, actually, there is one kind of test that isn't relevant: connection timeouts, or any error during connection setup (e.g. TLS).

So I'd only do it if it's retrying something that isn't occurring during connection setup.